src/utils/entityStore.ts

```diff
@@ -60,36 +85,58 @@ export interface IEntityStore<T extends IPositionedEntity> {
 export class EntityStore<T extends IPositionedEntity> implements IEntityStore<T> {
   private _entities: T[];
   private _tree: d3.Quadtree<T>;
+  private _rtree: RTree<T>;
 
   constructor() {
     this._entities = [];
+    this._rtree = new RTree<T>();
     this._tree = d3.quadtree<T>()
```

Comment:
nit: we can probably implement "nearest entity to point" with the rtree and get rid of the quadtree completely

Reply:
it's not exactly the same. the quadtree structure allows us to guarantee that no other point is closer based on the bounding box of the tree node. r-trees don't make the same guarantees, but they are great for querying overlapping bounds!